File: Data Structures & Algorithms/maximum-product-subarray/submission-1.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int mx=1,mn=1,ans=INT_MIN;
        for(int i=0;i<nums.size();i++){
            mx*=nums[i];
            mn*=nums[i];
            ans=max(ans,mx);
            ans=max(ans,mn);
            if(mx<=0) mx=1;
            if(mn==0) mn=1;
        }
        mx=1,mn=1;
        reverse(nums.begin(),nums.end());
        for(int i=0;i<nums.size();i++){
            mx*=nums[i];
            mn*=nums[i];
            ans=max(ans,mx);
            ans=max(ans,mn);
            if(mx<=0) mx=1;
            if(mn==0) mn=1;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
from typing import List
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
     ans=nums[0]
     n=len(nums)
     curr=0
     for i in range(0,n) :
        curr+=nums[i]
        ans=max(ans,curr)
        if(curr<0): curr=0
     return ans
    
       
